packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -245,28 +245,38 @@ impl PostgresDatabase {
     where
         T: DatabaseType + Serialize + Sync,
     {
-        sqlx::query(&format!(
+        // SAFETY: We insert a table name here, but `T::table()` is only accessible from within this
+        //   module.
+        let (inserted_id,) = sqlx::query_as(&format!(
             r#"
             INSERT INTO {} (
                 version_id,
                 schema,
                 created_by
             ) 
-            VALUES ($1, $2, $3);
+            VALUES ($1, $2, $3)
+            RETURNING version_id;
             "#,
             T::table()
         ))
         .bind(version_id)
         .bind(
             serde_json::to_value(database_type)
-                .unwrap_or_else(|err| unreachable!("Could not serialize data type: {err}")),
+                .report()
+                .change_context(InsertionError)?,
         )
         .bind(created_by)
-        .execute(&self.pool)
+        .fetch_one(&self.pool)
         .await
         .report()
         .change_context(InsertionError)?;
 
+        ensure!(
+            version_id == inserted_id,
+            Report::new(InsertionError)
+                .attach_printable("Inserted data does not match the expected data")
```

Comment:
What data is not matching, just the `id` or something else?

Reply:
Normally, this should never fail, as `sqlx` is required to return something.

Reply:
This is just for double checking, that we actually inserted the data. If the inserted version id does not match the expected one, we have bigger problems. I intentionally did not put more information on the attachment as this is probably not the source of the error but the implementation of the driver.